modules/tinymce/src/themes/silver/main/ts/ui/image/Images.ts

```diff
@@ -0,0 +1,70 @@
+import { AddEventsBehaviour, AlloyEvents, Behaviour, SimpleSpec } from '@ephox/alloy';
+import { Class, Insert, Ready, Remove, SelectorFind, SugarElement } from '@ephox/sugar';
+
+export type ImageProvider = () => Record<string, string>;
+
+interface ImageSpec {
+  readonly tag: string;
+  readonly classes: string[];
+  readonly attributes?: Record<string, string>;
+  readonly behaviours?: Array<Behaviour.NamedConfiguredBehaviour<any, any, any>>;
+}
+
+const renderImage = (spec: ImageSpec, imageUrl: string): SimpleSpec => {
+  const spinnerElement = SugarElement.fromTag('div');
+  Class.add(spinnerElement, 'tox-image-selector-loading-spinner');
+
+  const addSpinnerElement = (loadingElement: SugarElement) => {
+    Class.add(loadingElement, 'tox-image-selector-loading-spinner-wrapper');
+
+    Insert.append(loadingElement, spinnerElement);
+  };
+
+  const removeSpinnerElement = (loadingElement: SugarElement) => {
+    Class.remove(loadingElement, 'tox-image-selector-loading-spinner-wrapper');
+    Remove.remove(spinnerElement);
+  };
+
+  return {
+    dom: {
+      tag: spec.tag,
+      attributes: spec.attributes ?? {},
+      classes: spec.classes,
+    },
+    components: [{
+      dom: {
+        tag: 'div',
+        classes: [ 'tox-image-selector-image-wrapper' ]
+      },
+      components: [
+        {
+          dom: {
+            tag: 'img',
+            attributes: { src: imageUrl },
+            classes: [ 'tox-image-selector-image-img' ]
+          }
+        }
+      ]
+    }],
+    behaviours: Behaviour.derive([
+      ...spec.behaviours ?? [],
+      AddEventsBehaviour.config('render-image-events', [
+        AlloyEvents.runOnAttached((component) => {
+          addSpinnerElement(component.element);
+          SelectorFind.descendant<HTMLImageElement>(component.element, 'img').each((image) => {
+            Ready.image(image).finally(() => {
+              removeSpinnerElement(component.element);
+            }).catch(() => {
+              removeSpinnerElement(component.element);
+            });
```

Comment:
I'd rather have the comment to explicitly ignore the floating promise than a duplicate `catch` block to satisfy the linter. This is arguably worse than the `then/catch` version because on an error it will remove twice (not that this should throw an error, it's just unnecessary).

Reply:
changed, now we catch the error and remove the spinner only once